geopandas/geodataframe.py

```diff
@@ -600,6 +658,14 @@ def __getitem__(self, key):
             result.__class__ = DataFrame
         return result
 
+    def __setitem__(self, key, value):
+        if key == self._geometry_column_name:
```

Comment:
Can you add a comment / docstring here explaining why we are doing this here (why setitem is overridden), i.e. to try to preserve some crs in case someone did `df['geometry'] = [geom... for geom in df.geometry]`